scripts/Packages/Package_BGA/ipc_bga_generator.py

```diff
@@ -218,6 +253,12 @@ def rowNameGenerator(seq):
         # except yaml.YAMLError as exc:
             # print(exc)
     
+    with open(args.ipc_doc, 'r') as config_stream:
+        try:
+            configuration.update(yaml.safe_load(config_stream))
+        except yaml.YAMLError as exc:
+            print(exc)
```

Comment:
This exception should generate a better error message.
* File not found/not readable
* Yaml parsing error
Also the script should exit here.

The same applies to the block that reads the config file. Since both of those blocks basically do the same I suggest to refactor them into a function.
```python
for fn in [args.ipc_doc, ags.global_config]:
  update_configuration_from_yaml(fn)
`

Reply:
This applies to all the IPC generator scripts, and in fact I think many non-IPC generator scripts. In fact, I think it falls into the purview of https://github.com/pointhi/kicad-footprint-generator/issues/55, and would best be left as a bigger exercise once there is an agreed-upon direction for how to interface with our scripts. Then figuring out what errors can result and implementing them can happen uniformly.